packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,322 @@
+        if 'myid' in tar_info.name:
+            return None
+        if 'zookeeper.out' in tar_info.name:
+            return None
+        return tar_info
+
+    with tarfile.open(name=backup, mode='x:gz') as tar:
+        tar.add(
+            name=str(tmp_zookeeper_dir),
+            arcname='./zookeeper',
+            filter=_tar_filter,
+        )
+        if verbose:
+            tar.list()
+
+    print('Deleting temporary files in {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    shutil.rmtree(path=str(tmp_zookeeper_dir), ignore_errors=True)
+
+    print('ZooKeeper backup taken successfully')
+
+
+def restore_zookeeper(backup: Path, tmp_dir: Path, verbose: bool) -> None:
+    """
+    DC/OS ZooKeeper restore from backup procedure.
+
+    ZooKeeper changes files while it is running.
+    In order to have a consistent restore, one must stop ZooKeeper on all master nodes,
+    execute this procedure on all master nodes, and then start ZooKeeper again on all master nodes.
+
+    Stopping ZooKeeper on all master nodes inevitably causes the DC/OS cluster to experience downtime.
+    The restore procedure is intended for recovering from an unusable DC/OS cluster as a last resort measure.
+
+    Running this script requires at least as much free space as the ZooKeeper backup takes.
+    """
+    zookeeper_dir = EXHIBITOR_DIR / 'zookeeper'
+    tmp_zookeeper_dir = tmp_dir / 'zookeeper'
+
+    print('Restoring local ZooKeeper instance from {backup}'.format(backup=backup))
+
+    print('Moving ZooKeeper files temporarily to {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    # We copy files rather than move them so that if this script is interrupted
+    # while running, no data is lost inplace.
+    # However, this has a downside that we may use a lot of disk space -
+    # we have seen ZooKeeper files up to 20 GB in size.
+    _copy_dir_and_preserve_ownership(
+        src=zookeeper_dir,
+        dst=tmp_zookeeper_dir,
+        verbose=verbose,
+    )
+    shutil.rmtree(path=str(zookeeper_dir), ignore_errors=True)
+
+    print('Restoring {zookeeper_dir} from backup {backup}'.format(
+        zookeeper_dir=zookeeper_dir,
+        backup=backup,
+    ))
+    with tarfile.open(name=backup, mode='r:gz') as tar:
+        tar.extractall(path=str(EXHIBITOR_DIR))
+        if verbose:
+            tar.list()
+
+    print('Deleting temporary files in {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    shutil.rmtree(path=str(tmp_zookeeper_dir), ignore_errors=True)
+
+    print('Local ZooKeeper instance restored successfully')
+
+
+def _non_existing_file_path_existing_parent_dir(value) -> Path:
+    """
+    Validate that the value is a path to a file which does not exist
+    but the parent directory tree exists.
+    """
+    path = Path(value)
+    if path.exists():
+        raise ArgumentTypeError('Error: {} already exists'.format(path))
```

Comment:
Do these messages need `Error:` prefix? How does the final error message presented to a user look like?

Reply:
`error: argument backup_path: Error: sessions already exists`
Removing `Error` then :+1: